### ISSUE TITLE
Visual Studio doesn't open cshtml files

### ISSUE THREAD

**Issue by evorine** [aspnet/Tooling#276]
Firstly, my problem was the same as the problem mentioned here: https://github.com/aspnet/Tooling/issues/260

But after the restart, my Visual Studio stopped opening any .cshtml file. It just ignores my double clicks on the Solution Explorer. I can't even drag the file from the Windows Explorer if the file is from the same project that is loaded in the Visual Studio.

When I try to drag the file it says:

> The system cannot find the path specified. (Exception from HRESULT 0x80070003)

But I can drag and drop a cshtml if it is not included in the solution.

I tried to load it by right clicking and selecting **Open With**. I choose **HTML Editor (Default)**. Then I receive this error:

> The operation could not be completed.

**Note:** *Beta8 tools are installed*

**Comment by mgeorgedeveloper** [comment]
Same problem, but after installing ASP.NET 5 RC1.
Have you been able to resolve this?

**Comment by evorine** [comment]
Mine is solved after installing ASP.NET 5 RC1 and its web tools for Visual Studio.

**Comment by funkycoding** [comment]
Similar problem i guess, when i'm in a MVC6 project i can edit the *.cshtml files. When i'm in a MVC5 project and opening a cshtml file, it gives the following error in the activitylog.xml:

`
System.NullReferenceException: Object reference not set to an instance of an object. at Microsoft.VisualStudio.Web.Editors.Razor.DesignTimeHost.RazorDnxCommandInvoker.InvokeNoProtocolAdded(String cmdLine) at Microsoft.VisualStudio.Web.Editors.Razor.DesignTimeHost.RazorDnxCommandInvoker.Initialize(ITextBuffer textBuffer, IDnxCommandInvoker dnxCommandInvoker) at Microsoft.VisualStudio.Web.Editors.Razor.DesignTimeHost.RazorDnxCommandInvoker.Create(ITextBuffer textBuffer) at Microsoft.VisualStudio.Web.Editors.Razor.RazorCodeGenerator..ctor(ITextBuffer buffer, Version razorVersion, String physicalPath, String virtualPath) at Microsoft.VisualStudio.Web.Editors.Razor.RazorCodeGenerator.Create(ITextBuffer buffer, Version razorVersion, String physicalPath, String virtualPath) at Microsoft.VisualStudio.Web.Editors.Razor.RazorCodeGeneratorProvider.Microsoft.Html.Editor.Razor.IRazorCodeGeneratorProvider.CreateRazorCodeGenerator(ITextBuffer buffer, Version razorVersion, String physicalPath, String virtualPath) at Microsoft.Html.Editor.Razor.RazorCodeGeneratorFactory.CreateCodeGenerator(ITextBuffer buffer, Version razorVersion, String physicalPath, String virtualPath) at Microsoft.Html.Editor.ContentType.Handlers.RazorContentTypeHandler.Init(HtmlEditorTree editorTree) at Microsoft.Html.Editor.Document.HtmlEditorDocument..ctor(ITextBuffer textBuffer, IWebWorkspaceItem workspaceItem, Boolean disableContainedLanguages) at Microsoft.VisualStudio.Html.Package.Document.VsHtmlEditorDocument..ctor(IWebEditorInstance editorInstance, Boolean disableContainedLanguages) at Microsoft.VisualStudio.Html.Package.Commands.Html.VsHtmlTextViewConnectionListener.HtmlDocumentFactoryProjectionsDisabled.CreateDocument(IWebEditorInstance editorInstance) at Microsoft.Html.Editor.EditorFactory.HtmlEditorInstance..ctor(IWebWorkspaceItem workspaceItem, ITextBuffer diskBuffer, IWebEditorDocumentFactory documentFactory) at Microsoft.Html.Editor.EditorFactory.HtmlEditorFactory.CreateEditorInstance(IWebWorkspaceItem workspaceItem, Object textBuffer, IWebEditorDocumentFactory documentFactory) at Microsoft.VisualStudio.Html.Package.Commands.Html.VsHtmlTextViewConnectionListener.InitEditorInstance(ITextBuffer textBuffer) at Microsoft.VisualStudio.Html.Package.Commands.Html.VsHtmlTextViewConnectionListener.OnTextBufferCreated(ITextBuffer textBuffer) at Microsoft.Web.Editor.Controller.TextViewConnectionListener.SubjectBuffersConnected(IWpfTextView textView, ConnectionReason reason, Collection'1 subjectBuffers) at Microsoft.VisualStudio.Text.Editor.Implementation.ConnectionManager.<>c__DisplayClass4_1.<.ctor>b__1() at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.CallExtensionPoint(Object errorSource, Action call)
`

for completeness, running dnvm list
```
       1.0.0-rc1-final clr     x64          win
   *   1.0.0-rc1-final clr     x86          win             default
       1.0.0-rc1-final coreclr x64          win
       1.0.0-rc1-final coreclr x86          win
```

**Comment by mgeorgedeveloper** [comment]
Solved by installing Visual Studio Update 1 RC after it was initially broken when I first installed ASP.NET 5 RC1

**Comment by funkycoding** [comment]
@mgeorgedeveloper your solutions works! Thanks for the tip. I've installed [Visual Studio 2015 update rc1](https://www.visualstudio.com/en-us/news/vs2015-update1-vs.aspx) and all is working again.

**Comment by wmeints** [comment]
:+1: Works like a charm!

**Comment by sayedihashimi** [comment]
Sorry for the problems here. From the comments it looks like this is fixed in RC1 so I will go ahead and close this. If this is still an issue please let me know and we can reopen this.

**Comment by CodeChief** [comment]
I have the same problem AFTER installing VS 2015 Update 1 RTM! So that is not the fix. Same on multiple PCs. But I will try to find the current MVC package and see if it can be re-installed somehow. Activity log attached, remove the ".txt" suffix on both files to view.
[ActivityLog.xml.txt](https://github.com/aspnet/Tooling/files/61264/ActivityLog.xml.txt)
[ActivityLog.xsl.txt](https://github.com/aspnet/Tooling/files/61265/ActivityLog.xsl.txt)

**Comment by CodeChief** [comment]
Tried repairing the "Microsoft ASP.NET 5 RC1 Update 1" but it didn't fix the problem. Worse it won't allow itself to be uninstalled and the current download will not install because it's already installed. My development VM is trashed. Have to re-create the image again. This was a pure Update 1 image I only made last week, trashed in one week with current production "updates" (no pre-release option). I'm blocked on this. Have to waste more time with a new VM image, got deadlines to meet, not happy! Why do we have "release candidate" installs on production software?!?!?!!! @MSFT please make only stable, fully tested releases,,. because lots of other issues with VS 2015 Update 1 makes me wonder what happened with this one.

**Comment by fspezi** [comment]
@CodeChief, I have the same issue.
I updated VS2015 this weekend. Today I discover the bad bug.

**Comment by fspezi** [comment]
I found temporary solution to restore Visual Studio Razor editor:
* go to Control Panel, then uninstall "Microsoft ASP.NET 5 RC1 Update1"
* go to Control Panel, then reinstall Visual Studio: check "Web Developer Tools" options, then Install.

Done.

Now, you can open Visual Studio 2015, then open a .cshtml file.  
Warning: Visual Studio alert you that there is a update: "Microsoft ASP.NET and Web Tools". It seem that it is "Microsoft ASP.NET 5 RC1 Update1"; then: do not update it.

**Comment by BryanTheCrow** [comment]
@sayedihashimi This bug needs to be re-opened. I'm running into the same problem after updating to VS2015 Update 1. No .cshtml files will open. I never installed any RCs (...unless the RTM version of VS2015 Update 1 included an RC as part of it).

The error I get when trying to open .cshtml views:

"The operation could not be completed. Invalid pointer".

I'm trying @fspezi 's work-around now.

**Comment by swellfr** [comment]
A less intrusive solution than fspezi's one is:

1. Close VS
2. Delete the content of %LocalAppData%\Microsoft\VisualStudio\14.0\ComponentModelCache
3. Open VS

**Comment by fspezi** [comment]
My "%LocalAppData%\Microsoft\VisualStudio\14.0" path does not contains "ComponentModelCache" folder.  
Is it managed by "VS2015 Update 1" ?

**Comment by swellfr** [comment]
I'm using VS2015 Update 15 Community Edition with Nothing special

**Comment by fspezi** [comment]
I am using the same version, with some VS extension.

**Comment by logitron** [comment]
I was having this issue.  Followed steps by @fspezi: Uninstalled 'Microsoft ASP.NET 5 RC1 Update1' and after that, I could access the cshtml file again.  Then I installed 'Microsoft ASP.NET 5 RC1 Update1' back and, once again, couldn't access cshtml files.

So from that point, I proceeded with @swellfr's steps to delete the ComponentModelCache contents.  This seems to have resolved the issue for me.

**Comment by CodeChief** [comment]
Yes!!! Fixed it for me too. Thanks very much @fspezi 
Looks like the caching component has a bug which should be fixed ASAP.
Now is that Visual Studio's fault (MS Connect issue) or ASP.NET (this issue)? I've saved my corrupt component cache (attached) In case anyone wants to analyze.

Interestingly there is no content in the corrupt ".err" file. But there are a load of errors in the clean ".err" file, the one generated after a successful start and test opening a CSHTML file without errors. Look at the ".err" file in the "After Clean Started Successfully" subdirectory of my attached ZIP.

[20151223 - Visual Studio 2015 Update 1 corrupt component model cache.zip](https://github.com/aspnet/Tooling/files/70542/20151223.-.Visual.Studio.2015.Update.1.corrupt.component.model.cache.zip)

**Comment by CodeChief** [comment]
----- Catalog construction errors -----

Error #1
Microsoft.VisualStudio.Composition.PartDiscoveryException: Unable to load assembly "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\PrivateAssemblies\Microsoft.CodeAnalysis.VisualBasic.InteractiveEditorFeatures.dll" for scanning. ---> 

Error #2
Microsoft.VisualStudio.Composition.PartDiscoveryException: Unable to load assembly "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\PrivateAssemblies\Microsoft.VisualStudio.VisualBasic.Repl.dll" for scanning. ---> System.IO.FileNotFoundException: Could not load file or assembly 

Error #3
Microsoft.VisualStudio.Composition.PartDiscoveryException: Unable to load assembly "C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO 14.0\COMMON7\IDE\EXTENSIONS\MICROSOFT\WINDOWS AZURE TOOLS DIAGNOSTICS 1.0\Microsoft.VisualStudio.WindowsAzure.AzureDiagnostics.10Support.dll" for scanning. ---> System.IO.FileLoadException: Could not load file or assembly 

My errors stem from the Visual Basic Code Analysis, which seems to be part of a open/shared source http://source.roslyn.io/ project and the Azure tools.

I don't need either of those components, shame the VS 2015 installer makes a load of components mandatory, even if you never plan to use them.

**Comment by CodeChief** [comment]
After reading this http://blogs.msdn.com/b/webdev/archive/2015/12/07/new-feature-to-enable-c-6-vb-14.aspx I wonder if the problem is caused by trying to edit "older" projects which did not include the new "Roslyn" C#6/VB14 NuGet Package.

**Comment by logitron** [comment]
This issue happened for me after upgrading to the latest 'GitHub Extension for Visual Studio' v1.0.16.2.  I uninstalled the extension and performed the steps outlined by @fspezi to correct the issue; however, right after installing the extension back, the issue occurred again.  Can others who have had this issue confirm this?

**Comment by CodeChief** [comment]
I second that, about the same time I had troubles with my GitHub extension. I've fixed my CSHTML editing with the same solution, but GitHub is totally broken. Had to install the full desktop version to do any commits.

The behaviour I see (after the "fix" = workaround) is I try to login to "GitHub" then the logon dialog switches you over to "GitHub Enterprise" tab in the same dialog and disables the "GitHub" login. My account seems normal and the desktop version works. Perhaps it needs some of the "corrupt" cache entries and is silly enough not to re-initialize them.

Can nobody analyse the cache files I posted. In there should be some bad data and if there is any way to see the extension responsible then we know the answer. Besides external components corrupting the cache and making their own fix, don't forget the ASP.NET/Razor VS extension must also be fixed to not bomb-out this way.

**Comment by sameemd** [comment]
Thanks to @fspezi , was fighting this issue for 2 days!!

**Comment by eric-king** [comment]
This happened to me today also. Opened VS (2015 Enterprise Update 1) to a project that was working fine yesterday, pulled another dev's changes into my working copy, and couldn't open .cshtml files. 

Followed @swellfr 's  [instructions](https://github.com/aspnet/Tooling/issues/276#issuecomment-166650817) to delete the component model cache, now working fine.

**Comment by jphellemons** [comment]
Followed @swellfr instructions worked for me. No need to uninstall and reinstall the rc1 update 1. Just removed the files in the 'ComponentModelCache' folder.

**Comment by Hayke** [comment]
@swellfr Thanks, that solved it for me. Empty the ComponentModelCache folder was the solution.

**Comment by drinkbird** [comment]
@swellfr Thanks, your solution works perfectly.

**Comment by JohannesEH** [comment]
Still doesn't work for me after deleting component cache content.

**Comment by madmox** [comment]
Doesn't work for me either after deleting the component cache (even with [this extension](http://vsixgallery.com/extension/f5028141-9dd0-4ac4-ae6d-0481ae9a940d/)).

**Comment by guidobouman** [comment]
@swellfr's fix worked for me: https://github.com/aspnet/Tooling/issues/276#issuecomment-166650817

**Comment by cemdev** [comment]
deleting the .err file also worked for me - until I closed and opened VS again, then it was back.... Seems like I have to delete it every time I re-start VS/the machine. Really annoying, guess I'll have to write a startup script to delete it every time.

**Comment by kooldave98** [comment]
Just got this problem today, after updating my Github extension for VS. 

Deleting the component model cache folder contents as stated above fixed it for me.

**Comment by guidobouman** [comment]
@cemdev Did you close your VS before deleting the files?

**Comment by cemdev** [comment]
I did, yes.

**Comment by yooakim** [comment]
I also stumbled upon this issue today. Everything had been working fine. yesterday I updated the Github plugin (as prompted by VS2015) - after a reboot I could no longer open cshtml files.

For me running this from an administrative prompt fixe the issue:

```
C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\devenv.exe /resetuserdata
```

**Comment by GearTheWorld** [comment]
Ouffff what a nightmare.  2.5 hours trying to figure out this mess to arrive here and trying things.  At least deleting the content of the folder ComponentModelCache  solves one thing now compiling my project gives 36 errors all about System.Web not found however it is there so what is going on with this second problem is still a mystery to me.

**Comment by GearTheWorld** [comment]
One of the 36 errors I got : 

Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0234	The type or namespace name 'Optimization' does not exist in the namespace 'System.Web' (are you missing an assembly reference?)C:\...............\App_Start\BundleConfig.cs

**Comment by DominikErnst** [comment]
I fixed my VS 2015 Update 1 with:
Right click on a CSHTML - File and set HTML (**Web Forms**) Editor as Default

**Comment by FabienLavocat** [comment]
@DominikErnst thanks, I just got the same error today, for no reason. I didn't install anything new.

**Comment by DominikErnst** [comment]
VS2015 is a mess, I hope there is soon a Update 2.

**Comment by jasel-lewis** [comment]
@DominikErnst - thank you.  I ran up against the same issue this morning after allowing VS2015 Enterprise to update Application Insights Tools (and I believe there was a core update to VS2015 that was run as well - possibly Update 1??).  Your solution allows me to open CSHTML files now, but (and as one would expect), I've lost the Razor syntaxing.  Regardless - thank you for the quick fix!  I'll work on getting the syntaxing back when I have time - for now, code production be put on hold.

**Comment by guidobouman** [comment]
@jasel-lewis Read the whole thread. @swellfr's solution (https://github.com/aspnet/Tooling/issues/276#issuecomment-166650817) fixes the issue completely.

@DominikErnst I don't think VS2015 is a mess, it's better than VS2013 in my opinion. They just released an update that doesn't rebuild caches and in turn breaks Razor in VS. Not tested well apparently.

**Comment by jasel-lewis** [comment]
@guidobouman I read the entire thread and tried different solutions before I posted, it's interesting to know you assume I had not.  No, @swellfr's proposal does not "fix[es] the issue completely" in my situation.  I have tried the solutions provided by @swellfr, @DominikErnst and @yooakim, in that order, and no luck getting back to the way VS was operating this morning with respect to .CSHTML files.  At least "forcing" a different editor gets me around the "does not point to an instance of an object" error.

**Comment by sayedihashimi** [comment]
I'm having a hard time following the thread but it looks like @jasel-lewis is still having some issues. Any one else having issues with this after trying (https://github.com/aspnet/Tooling/issues/276#issuecomment-166650817)?

@jasel-lewis (_and anyone else who still has the problem_). Can you paste the results of Help->About in VS 2015. I'd like to see what you have installed. We may need to open a new issue for this since it looks like it may be a different issue not related to the component model cache, but let's keep it here for now.

**Comment by jasel-lewis** [comment]
@sayedihashimi Please do not re-open for me.  My initial post in here was simply to thank @DominikErnst for his proposal as that provided a temporary fix for me.  I have resolved my issue through really nasty brute-force by completely uninstalling EVERYTHING VS2015-related (to include 3rd party installations and all earlier-era redistributables), reinstalling and then performing a less-brute-force version of @yooakim's proposal above ([comment 175040047](https://github.com/aspnet/Tooling/issues/276#issuecomment-175040047)) where I used the /ResetSettings flag (instead of /ResetUserData).

**Comment by DominikErnst** [comment]
@jasel-lewis Thanks, **devenv /ResetSettings** solved the issue completely.

**Comment by guidobouman** [comment]
@jasel-lewis Excuses me, I jumped the gun on that response. I assumed you had not read the thread as this fix worked for everyone in our office. Probably because we all use similar configurations, updates and extensions. My bad! Nice follow up on your next actions by the way.

**Comment by sayedihashimi** [comment]
@jasel-lewis sorry you ran into these issues, I'm happy you were able to get it resolved though.

**Comment by FabienLavocat** [comment]
**devenv /ResetSettings** solved the issue for me too, thanks a lot

**Comment by cemdev** [comment]
devenv /resetsettings did not fix it for me, but deleting the content of ComponentModelCached did (again). This is starting to get very annoying.

**Comment by wakeful-sun** [comment]
@swellfr Thanks, your solution solved my problem.

**Comment by Paxxi** [comment]
Happened the second time for me today with update 1 and rc1 installed. Updated an unrelated extension and suddenly .cshtml files were broken. Clearing ComponentCache solved the issue.

**Comment by DominikErnst** [comment]
Maybe the Servicing Update (KB3110221) for "Update 1" can fix this issue, try ...

https://blogs.msdn.microsoft.com/visualstudioalm/2016/01/26/servicing-update-available-for-visual-studio-update-1/

**Comment by bmolloy** [comment]
Thanks @swellfr  - that did the trick for me!

**Comment by mittork** [comment]
@sayedihashimi I am still having this issue. I tried everything mentioned here and in other similar issues... Nothing helps, nothing. 
Here the content from Help->About:

Microsoft Visual Studio Enterprise 2015
Version 14.0.24720.00 Update 1
Microsoft .NET Framework
Version 4.6.01038

Installed Version: Enterprise

Architecture and Modeling Tools   00322-80000-00000-AA202
Microsoft Architecture and Modeling Tools
    
UML® and Unified Modeling Language™ are trademarks or registered trademarks of the Object Management Group, Inc. in the United States and other countries.

Microsoft Visual Studio Tools for Applications 2015   00322-80000-00000-AA202
Microsoft Visual Studio Tools for Applications 2015

Visual Basic 2015   00322-80000-00000-AA202
Microsoft Visual Basic 2015

Visual C# 2015   00322-80000-00000-AA202
Microsoft Visual C# 2015

Visual C++ 2015   00322-80000-00000-AA202
Microsoft Visual C++ 2015

Visual F# 2015   00322-80000-00000-AA202
Microsoft Visual F# 2015

Application Insights Tools for Visual Studio Package   1.0
Application Insights Tools for Visual Studio

Arduino for Visual Studio   1.0
Arduino extension for Visual Studio. For more information please see docs, faq and forum at http://www.visualmicro.com

ASP.NET and Web Tools 2015 (RC1 Update 1)   14.1.20203.0
ASP.NET and Web Tools 2015 (RC1 Update 1)

Common Azure Tools   1.7
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

DataFactoryProject   1.0
Microsoft Data Factory Package

JetBrains ReSharper Ultimate 10.0.2   Build 104.0.20151218.120627
JetBrains ReSharper Ultimate package for Microsoft Visual Studio. For more information about ReSharper Ultimate, visit http://www.jetbrains.com/resharper. Copyright © 2016 JetBrains, Inc.

Microsoft Azure Data Factory Node Node   1.0
Azure Data Factory extension for Visual Studio Server Explorer.

Microsoft Azure HDInsight HQL Service   2.0.2900.0
Language service for Hive query

Microsoft Azure HDInsight Tools for Visual Studio   2.0.2900.0
An integrated development environment for HDInsight application development.

Microsoft Azure Mobile Services Tools   1.4
Microsoft Azure Mobile Services Tools

Microsoft Azure Tools   2.8
Microsoft Azure Tools for Microsoft Visual Studio 2015 - v2.8.31121.1

Microsoft Azure Tools   2.7
Microsoft Azure Tools for Microsoft Visual Studio 2015 - v2.7.30818.1601

Node.js Tools   1.1.31117.00
Adds support for developing and debugging Node.js apps in Visual Studio

Node.js Tools - Profiling   1.1.31117.00
Profiling support for Node.js projects.

NuGet Package Manager   3.3.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit http://docs.nuget.org/.

NuProjPackage   1.0
NuGet Package Project System

PHP Tools for Visual Studio   1.19.8608.2015
PHP Tools extend Visual Studio with a set of features to build PHP applications more efficiently. It makes code more readable, easier to navigate, and clean.

PreEmptive Analytics Visualizer   1.2
Microsoft Visual Studio extension to visualize aggregated summaries from the PreEmptive Analytics product.

Python Tools for Visual Studio   2.2.31124.00
Python Tools for Visual Studio provides IntelliSense, projects, templates, Interactive windows, and other support for Python developers.

Python Tools for Visual Studio - Django Integration   2.2.31124.00
Provides templates and integration for the Django web framework.

Python Tools for Visual Studio - Profiling Support   2.2.31124.00
Profiling support for Python projects.

SQL Server Analysis Services   13.0.800.111
Microsoft SQL Server Analysis Services Designer 
Version 13.0.800.111

SQL Server Data Tools   14.0.51128.0
Microsoft SQL Server Data Tools

SQL Server Integration Services   
Microsoft SQL Server Integration Services Designer
Version 13.0.800.111

SQL Server Reporting Services   13.0.800.111
Microsoft SQL Server Reporting Services Designers 
Version 13.0.800.111

TypeScript   1.7.6.0
TypeScript for Microsoft Visual Studio

**Comment by ccasalicchio** [comment]
@swellfr workaround did it for me.

> Close VS
> Delete the content of %LocalAppData%\Microsoft\VisualStudio\14.0\ComponentModelCache
> Open VS

Thanks

**Comment by piersh** [comment]
I had the same problem. deleting ComponentModelCache fixed it.

**Comment by bandersen22000** [comment]
I have the same error and I have tried the following with NO SUCCES

- Delete the content of %LocalAppData%\Microsoft\VisualStudio\14.0\ComponentModelCache
- “%ProgramFiles%\Common Files\Microsoft Shared\VS7DEBUG” to something else. I changed it to VS7DEBUG1.
- Installed Servicing Update (KB3110221) - https://blogs.msdn.microsoft.com/visualstudioalm/2016/01/26/servicing-update-available-for-visual-studio-update-1/
- C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\devenv.exe /resetuserdata
- C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\devenv.exe /resetsettings 

**I finally uninstalled visual studio 2015. Restarted and installed it again and then it worked.**

**Comment by beppler** [comment]
For me remove component cache did not solve the problem.
I tried the resetuserdata and resetsettings with no luck too.
Even tried the Servicing Update and the problem persists.

**Comment by mikerouse** [comment]
This problem happened for me when I installed Web Essentials 2015.1 version 1.0.209. Deleting ComponentModelCache fixed it. 

> Close VS
> Delete the content of %LocalAppData%\Microsoft\VisualStudio\14.0\ComponentModelCache
> Open VS

Thanks @swellfr

**Comment by jamsoft** [comment]
Yup, I'm going round and round on this issue as well.  Multiple restarts  day required and I'm loathed to install any updates as this seems to really bork things for people.  This seems to be a very wide-spread problem.  I see the same .err file issues as @CodeChief I've lost count of the number of times my ComponentModelCache has been deleted and the problem ALWAYS comes back again.

I also see this in my ActivityLog.xml file:

```
  <entry>
    <record>639</record>
    <time>2016/03/04 09:47:00.008</time>
    <type>Error</type>
    <source>Microsoft.VisualStudio.CommonIDE.ExtensibilityHosting.VsShellComponentModelHost</source>
    <description>Assembly timestamp in MEF Cache does not match loaded assembly timestamp;VS.Platform.ExtensibilityHosting.MefCacheInvalidation.Description=Assembly timestamp in MEF Cache does not match loaded assembly timestamp;VS.Platform.ExtensibilityHosting.MefCacheInvalidation.CachedAssembly.Name=Microsoft.VisualStudio.WebClient.Diagnostics.Services, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a;VS.Platform.ExtensibilityHosting.MefCacheInvalidation.CachedAssembly.Path=C:/Program%20Files%20(x86)/Microsoft%20Visual%20Studio%2014.0/Common7/IDE/PrivateAssemblies/Microsoft.VisualStudio.WebClient.Diagnostics.Services.DLL;VS.Platform.ExtensibilityHosting.MefCacheInvalidation.CachedAssembly.Timestamp=19/11/2015 21:31:16;VS.Platform.ExtensibilityHosting.MefCacheInvalidation.LoadedAssembly.Name=Microsoft.VisualStudio.WebClient.Diagnostics.Services, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a;VS.Platform.ExtensibilityHosting.MefCacheInvalidation.LoadedAssembly.Path=C:/Program%20Files%20(x86)/Microsoft%20Visual%20Studio%2014.0/Common7/IDE/PrivateAssemblies/Microsoft.VisualStudio.WebClient.Diagnostics.Services.DLL;VS.Platform.ExtensibilityHosting.MefCacheInvalidation.LoadedAssembly.Timestamp=02/03/2016 20:15:09&#x000D;&#x000A;</description>
  </entry>
```

```
  <entry>
    <record>641</record>
    <time>2016/03/04 09:47:00.014</time>
    <type>Error</type>
    <source>VisualStudio</source>
    <description>End package load [Microsoft.VisualStudio.WebClient.Diagnostics.HtmlToolHost.Package.HtmlToolHostPackage, Microsoft.VisualStudio.WebClient.Diagnostics.HtmlToolHost, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a]</description>
    <guid>{B66F381B-48DE-4C10-B71B-EF9A2A808D85}</guid>
    <hr>80131500</hr>
    <errorinfo>The Visual Studio component cache is out of date. Please restart Visual Studio. The Visual Studio component cache is out of date. Please restart Visual Studio.</errorinfo>
  </entry>
```

**Comment by jamsoft** [comment]
How can this be closed with so many people still having issues?

**Comment by mttmccb** [comment]
I've been struggling with this, nothing I've tried has worked, I'm hoping it doesn't impact the project build.

**Comment by mttmccb** [comment]
@sayedihashimi Here's my about info

Microsoft Visual Studio Professional 2015
Version 14.0.24720.00 Update 1
Microsoft .NET Framework
Version 4.6.01055

Installed Version: Professional

Microsoft Visual Studio Tools for Applications 2015   00322-40000-00000-AA800
Microsoft Visual Studio Tools for Applications 2015

Visual Basic 2015   00322-40000-00000-AA800
Microsoft Visual Basic 2015

Visual C# 2015   00322-40000-00000-AA800
Microsoft Visual C# 2015

Visual C++ 2015   00322-40000-00000-AA800
Microsoft Visual C++ 2015

Application Insights Tools for Visual Studio Package   4.2.60128.3
Application Insights Tools for Visual Studio

ASP.NET and Web Tools 2015 (RC1 Update 1)   14.1.20203.0
ASP.NET and Web Tools 2015 (RC1 Update 1)

Azure App Service Tools v2.8.2   14.0.20201.0
Azure App Service Tools v2.8.2

Common Azure Tools   1.7
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

DataFactoryProject   1.0
Microsoft Data Factory Package

GitHub.VisualStudio   1.0
A Visual Studio Extension that brings the GitHub Flow into Visual Studio.

Microsoft Azure Data Factory Node Node   1.0
Azure Data Factory extension for Visual Studio Server Explorer.

Microsoft Azure HDInsight HQL Service   2.0.2900.0
Language service for Hive query

Microsoft Azure HDInsight Tools for Visual Studio   2.0.2900.0
An integrated development environment for HDInsight application development.

Microsoft Azure Mobile Services Tools   1.4
Microsoft Azure Mobile Services Tools

Microsoft Azure Tools   2.7
Microsoft Azure Tools for Microsoft Visual Studio 2015 - v2.7.30728.1602

Microsoft Azure Tools   2.8
Microsoft Azure Tools for Microsoft Visual Studio 2015 - v2.8.40211.2

Node.js Tools   1.1.30716.01
Adds support for developing and debugging Node.js apps in Visual Studio

Node.js Tools - Profiling   1.1.30716.01
Profiling support for Node.js projects.

NuGet Package Manager   3.3.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit http://docs.nuget.org/.

PreEmptive Analytics Visualizer   1.2
Microsoft Visual Studio extension to visualize aggregated summaries from the PreEmptive Analytics product.

RoamedExtensionVSPackage Extension   1.0
RoamedExtensionVSPackage Visual Studio Extension Detailed Info

SQL Server Analysis Services   13.0.1000.281
Microsoft SQL Server Analysis Services Designer 
Version 13.0.1000.281

SQL Server Data Tools   14.0.60203.0
Microsoft SQL Server Data Tools

SQL Server Integration Services   
Microsoft SQL Server Integration Services Designer
Version 13.0.1000.281

SQL Server Reporting Services   13.0.1000.281
Microsoft SQL Server Reporting Services Designers 
Version 13.0.1000.281

TypeScript   1.7.6.0
TypeScript for Microsoft Visual Studio

**Comment by dove** [comment]
thanks @swellfr worked a treat.

**Comment by cemdev** [comment]
Yep, I still have to delete the componentcache files every time I restart my laptop.

**Comment by jpjohnson** [comment]
@sayedihashimi, just had this issue today. VS had all updates applied before the issue. Deleting the files under `ComponentModelCache` directory seemed to fix the issue.

**Comment by marinade** [comment]
Had this issue today as well. 
**devenv /resetuserdata** worked for me.

**Comment by jamsoft** [comment]
@marinade I've tried that multiple times, and the other solutions I've found.  Nothing seem to fix it permanently.  So frustrating.

**Comment by jamsoft** [comment]
@marinade Just out of interest do you have any .err files in your component cache?

**Comment by marinade** [comment]
@jamsoft I do have one .err file called Microsoft.VisualStudio.Default.err which has a couple logging statements from Microsoft.AppInsights.GettingStarted.ZeroClickLogging.  I don't think this is related but I could be wrong...

**Comment by jamsoft** [comment]
@marinade Thanks for that info, I see those as well but then later on after a crash I get further errors in there about VisualBasic code analysis and Interaction problems.  But I've never used VB, nor code analysis, but my system event log crash report does include ReSharper in the stack trace ... interesting ... hmm ...

**Comment by clement911** [comment]
This started happening today after rebooting my machine.
https://github.com/aspnet/Tooling/issues/276#issuecomment-166650817 this worked for me.

**Comment by Craig0409** [comment]
I've had this same problem, and installing Visual Studio 2015 Update 2 solved my problem, FYI.

**Comment by cbmek** [comment]
Had the same issue (couldn't open *.cshtml files from MVC 5 after installin ASP.NET 5).
Installing [Visual Studio 2015 Update 2 RC](https://www.visualstudio.com/news/vs2015-update2-vs) solved the problem - I hope it didn't  :boom:  anything else.

**Comment by mttmccb** [comment]
Installing Visual Studio 2015 Update 2 RC is the only thing that fixed it for me.

**Comment by ComputerCowboy** [comment]
@fspezi Your solution worked, although it took a while
Thanks for the tip

**Comment by chuksogbolime** [comment]
I'm currently having same issue with update3. Can't seem to understand exactly what @GerjanOnline did to fix the issue. Can anyone help?

**Comment by mirhagk** [comment]
I had a similar issue (except complaining about invalid pointers) after Update 3. Installing Update `KB3165756` seems to have fixed the issue, so make sure you're fully updated

**Comment by bomzj** [comment]
VS 2015 -> Deleting all the files in the ComponentModelCache worked for me:

C:\Users\**username**\AppData\Local\Microsoft\VisualStudio\14.0\ComponentModelCache

**Comment by MehdiKarimi** [comment]
"The operation could not be completed. invalid pointer"
I have the same problem AFTER installing DotNetCore.1.0.1-VS2015Tools.Preview2.0.3
VS 2015 Update 3! So that is not the fix.

**Comment by PrimeByDesign** [comment]
@fspezi Your solution worked for me! Thank you.

**Comment by nkalfov** [comment]
I had the same issue, when opening a .cshtml file, VS would alert me "The operation could not be completed". I removed  DotNetCore.1.0.1-VS2015Tools.Preview2.0.3, and executed in CMD
` devenv /resetuserdata `. This fixed the issue for 2 days, but then I started getting the same error. Then I removed all azure extensions that I could and now I can open .cshtml files.

**Comment by chrismbarr** [comment]
This seems to work for me, I [found it on StackOverflow](http://stackoverflow.com/a/34329045/79677)

:one:  Close Visual Studio
:two: Delete the folder located at `%LocalAppData%\Microsoft\VisualStudio\14.0\ComponentModelCache`
:three: Re-open Visual Studio again

**Comment by AndrewH2O** [comment]
Exactly the same situation as chrismbarr - previously uninstalled web essentials and repaired Microsoft ASP.NET RC1 Update 1. This issue popped up at random today.

**Comment by jonasninja** [comment]
My reproduction of this issue was also seemingly at random. As far as I know, there were no updates or installations. Deleting `ComponentModelCache` works. Thanks, @swellfr!

**Comment by PrimeByDesign** [comment]
Thank you, this works for me.

**Comment by pchicociv** [comment]
Happened to me after instaling SonarLint extension.
Solved deleting the content of ComponentModelCache folder following the [instructions above](https://github.com/aspnet/Tooling/issues/276#issuecomment-166650817)

**Comment by MarkBird** [comment]
This just started happening to me, I haven't used Visual Studio 2015 for a while, deleting ComponentModelCache does not work for me.

**Comment by thecopypaster** [comment]
The issue was fixed after I enabled Razor Language Services extension.

**Comment by SanderPs** [comment]
I also got "The operation could not be completed. invalid pointer", but for me clearing ComponentModelCache did not work. 

**What did work was this**:
- i ran Visual Studio Installer by right-clicking it and choosing 'Run as Administrator'
- i clicked Modify
- in the window that opened that did not choose anything, but just clicked at the Modify button to close the window
- i then saw that some files where downloaded and installed
- when I launched VS2017 i could open .cshtml files normally again!

I am not sure 'Run as Administrator' made the difference, but apparently not everything got installed when I updated VS2017 to the latest version (i now have 15.9.11)

**Comment by barrytang** [comment]
To report new issues, use the [Report a Problem](https://developercommunity.visualstudio.com/) tool. The GitHub Issue Tracker for the aspnet/Tooling repo is now deprecated in favor of the Report a Problem tool.

If you encounter a problem with Visual Studio, we want to know about it so that we can diagnose and fix it. By using the Report a Problem tool, you can collect detailed information about the problem, and send it to Microsoft with just a few button clicks. See [here](https://docs.microsoft.com/en-us/visualstudio/ide/talk-to-us) for more details.
